packages/redux-query-react/src/hooks/use-request.js

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { useDispatch } from 'react-redux';
+import { requestAsync, cancelQuery, getQueryKey } from 'redux-query';
+
+import useConstCallback from './use-const-callback';
+import useMemoizedAction from './use-memoized-action';
+
+const useRequest = providedQueryConfig => {
+  const reduxDispatch = useDispatch();
+
+  const isPendingRef = React.useRef(false);
+
+  const [isPending, setIsPending] = React.useState(false);
+
+  const finishedCallback = useConstCallback(() => {
+    setIsPending(false);
+    isPendingRef.current = false;
+  });
+
+  const transformQueryConfigToAction = useConstCallback(queryConfig => {
+    return {
+      ...queryConfig,
+      unstable_preDispatchCallback: finishedCallback,
```

Comment:
I don't remember the significance of this. Something to do with trying to keep the props of the component sane with updates coming from different sources?

Does it still work and make sense in a hooks/concurrent world?

Reply:
It's to avoid sending cancelation requests in cases when the query was just finished.

If you don't use a sync callback and instead use promises, the `then` callback will trigger as a micro-task after the redux store updated. If React re-renders the component/hook between those two events and the query keys change, then the component/hook's internal state will still indicate that the query is still in-flight and therefore dispatch out a cancelation action.

I'm not confident how hooks or concurrent could impact this. My guess is it may change the race to make this not as much of an issue as React might be more likely to delay re-rendering. Given that, I think it's best to keep the callback as-is.

Tangential question – I wonder if we should make the "Trying to cancel a request that is not in flight" warning dev-only (or completely drop it?). It's not such a bad thing to dispatch a cancelation action, right?

Reply:
> Given that, I think it's best to keep the callback as-is.

👍 

> Tangential question – I wonder if we should make the "Trying to cancel a request that is not in flight" warning dev-only (or completely drop it?). It's not such a bad thing to dispatch a cancelation action, right?

Yea, that sounds fine.